ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobs.tsx

```diff
@@ -45,33 +57,88 @@ type ReportJobsProps = {
 };
 
 function ReportJobs({ scanConfig }: ReportJobsProps) {
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const [filteredStatuses, setFilteredStatuses] = useState<RunState[]>([]);
+    const [showOnlyMyJobs, setShowOnlyMyJobs] = React.useState<boolean>(false);
+
+    const onReportStatusFilterChange = (_checked: boolean, selectedStatus: RunState) => {
+        setFilteredStatuses((prevFilteredStatuses) => {
+            const isStatusIncluded = prevFilteredStatuses.includes(selectedStatus);
+            if (isStatusIncluded) {
+                return prevFilteredStatuses.filter((status) => status !== selectedStatus);
+            }
+            return [...prevFilteredStatuses, selectedStatus];
+        });
+        setPage(1);
+    };
+
+    const onMyJobsFilterToggle = (checked: boolean) => {
+        setShowOnlyMyJobs(checked);
+        setPage(1);
```

Comment:
Here is a question related to TODO comment below: is state of my jobs filter:
1. a parameter of the future request?
2. in URL parameters of page?

Especially if 2, then `setPage(1)` has become implicit whenever there is a change to URL parameters?

Reply:
For future request, be aware of https://github.com/stackrox/stackrox/pull/12728

Reply:
In VM Reports, we seem to keep the filter state in the local component state. We'll most likely do that here too. We switch between two different endpoints based on that state:

1. `/v2/compliance/scan/configuration/{id}/reports/history`
2. `/v2/compliance/scan/configuration/{id}/reports/my-history`

Reply:
Yeah, that is what caught my eye: `useURLPagination` hook versus local state.

From staging, it looks like **Configuration details** versus **All report jobs** is also in local state.
Off the subject, somewhat related to the switch, notice **All** in tab versus switch to filter **My**

Although it feels like all or none for local versus URL is safer, I will leave to your best judgment.

Maybe a good item for brainstorming cost and benefit within the team?

Reply:
On the topic of `All report jobs`. I guess the proper term should be `Report jobs` since we could show all or just yours.

Reply:
If I understand correctly, change from page state to page URL for Vulnerability Reports is consistent with minor update like 4.6.0 as far a semantic versions: incremental change toward consistency with usual behavior.

It could be mentioned in release notes (via label or whatever, if you create a separate follow up task to do it).

Reply:
Sounds good. I'll ask Product and UX about that little behavior change